### DISCUSSION TITLE
getDateDiff and years

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#14562] [General - Components / Directives / etc]
Hi

I need to have the difference between 2 dates in years unit so I'm using date.getDateDiff with 2 options 'year' and years'.

1. What is the difference between 'year' and 'years'? Both gave me the same result.

2. When using 'year', I see the following error
![image](https://user-images.githubusercontent.com/100950035/194089785-9cd06277-c5c1-48b3-8644-74e6e106740e.png)
which is coming from here: `function getDateDiff(date: Date | number | string, subtract: Date | number | string, unit?: `${DateUnitOptions}s`): number;` and based on this thread: https://github.com/quasarframework/quasar/issues/12616 it was done on purpose but I don't understand the logic of this additional 's' at the end, it causing 'year' to be 'years' and 'years' to be 'yearss'.

3. When asking for the difference in years, I want it to be the actual years so if the date is 2000-10-10, I expect 2022-10-09 to return 11 and 2022-10-11 to return 12. Is it possible and I missed something or this is not the intention of this method?

Thanks

**Comment by MilosPaunovic** [comment]
1. There is no difference between `year` and `years`, it handles them the same, see below for more info: https://github.com/quasarframework/quasar/blob/952e7d8ca86fe7550c51771c9d12cd92a3db3dde/ui/src/utils/date.js#L612

2. It does seem like a bug to my not TS trained eye, it was introduced in this commit https://github.com/quasarframework/quasar/commit/766cc763f00d404f930798addc0216b694335350, so maybe we could wait Paolo for input.

3. The `getDateDiff` helper is using subtraction between two `date.getFullYear` numbers, it does not take months/days into account, see below for more info: https://github.com/quasarframework/quasar/blob/952e7d8ca86fe7550c51771c9d12cd92a3db3dde/ui/src/utils/date.js#L615

**Comment by AmosSzust** [comment]
Thanks a lot for your reply.

1. So what is the reason for this duplication?

2. I think this is a bug and I wanted to post about it but then I saw the thread I posted before so I was surprised.

3. I see. Then this is not good for me. I wonder if I'll make a PR about this (and about month), it will be approved.

**Reply by pdanpdan** [reply]
In the beginning it was a mix of singulars and plurals. Then it was decided to be plurals all over, but for back-compat reasons the singular versions were still accepted. But the singulars are not and should not be advertised (like in TS types) because are deprecated.

**Comment by AmosSzust** [comment]
Thanks @pdanpdan 
Regarding point 3, may I create a PR regarding the years/months difference to make it calculate based on years only (as it is now, default for backward compatibility) or based on full date (additional parameter to the getDateDiff method)?

**Comment by pdanpdan** [comment]
I think @rstoenescu can decide this.
From my pov it would be nice, even if it doesn't get into quasar - at least it would be an example for who needs it.

**Reply by AmosSzust** [reply]
It would be helpful for me and maybe others in the future :) I'll wait for @rstoenescu decision then. Thanks.

**Comment by AmosSzust** [comment]
Hi @rstoenescu, what do you say? I prefer to do it after you will say this is acceptable by you so let me know. Thanks.
